packages/cli-repl/test/e2e.spec.ts

```diff
@@ -1143,5 +1142,120 @@ describe('e2e', function() {
+      }
+    });
+
+    it('returns a modified identifier for fn', async() => {
+      const shellOriginalInput = "const fn = function () { console.log(111); }; edit('fn')";
+      const editorOutput = `function () {
+        console.log(222);
+      };`;
+      const shellModifiedInput = 'fn = function () { console.log(222); };';
+      const editor = await fakeExternalEditor(editorOutput);
+      const result = await shell.executeLine(`config.set("editor", "${editor}");`);
+
+      expect(result).to.include('"editor" has been changed');
+      shell.writeInputLine(shellOriginalInput);
+      await eventually(() => {
+        shell.assertContainsOutput(shellModifiedInput);
+      });
+    });
+
+    it('returns a modified identifier for var', async() => {
+      const shellOriginalInput = "const myVar = '111'; edit('myVar')";
+      const editorOutput = "const myVar = '222';";
+      const shellModifiedInput = "myVar = '222';";
+      const editor = await fakeExternalEditor(editorOutput);
+      const result = await shell.executeLine(`config.set("editor", "${editor}");`);
+
+      expect(result).to.include('"editor" has been changed');
+      shell.writeInputLine(shellOriginalInput);
+      await eventually(() => {
+        shell.assertContainsOutput(shellModifiedInput);
+      });
+    });
+
+    it('returns a modified identifier for a.b.c', async() => {
+      const shellOriginalInput = "const myObj = { field: { child: 'string value' } }; edit('myObj')";
+      const editorOutput = `const myObj = {
+        field: {
+          child: 'new   value'
+        }
+      };`;
+      const shellModifiedInput = "myObj = { field: { child: 'new   value' } };";
+      const editor = await fakeExternalEditor(editorOutput);
+      const result = await shell.executeLine(`config.set("editor", "${editor}");`);
+
+      expect(result).to.include('"editor" has been changed');
+      shell.writeInputLine(shellOriginalInput);
+      await eventually(() => {
+        shell.assertContainsOutput(shellModifiedInput);
+      });
+    });
+
+    it('returns an error when editor exits with exitCode 1', async() => {
+      const shellOriginalInput = 'edit db.test.foobar';
+      const editor = await fakeExternalEditor();
+      const result = await shell.executeLine(`config.set("editor", "${editor}");`);
+
+      expect(result).to.include('"editor" has been changed');
+      shell.writeInputLine(shellOriginalInput);
+      await eventually(() => {
+        shell.assertContainsError('failed with an exit code 1');
+      });
+    });
+
+    it('opens an empty editor', async() => {
+      const output = '';
+      const editor = await fakeExternalEditor(output);
+      const result = await shell.executeLine(`config.set("editor", "${editor}");`);
+
+      expect(result).to.include('"editor" has been changed');
+      shell.writeInputLine('edit');
+      await eventually(() => {
+        shell.assertContainsOutput(output);
+      });
+    });
+  });
```

Comment:
We put the snippets tests into its own file, if you like you can do that here as well (especially since the editor tests here don’t connect to a database)